### DISCUSSION TITLE
Inline Content Component has unexpected structure

### DISCUSSION THREAD

**Discussion by JohannaPeanut** [Thinkmill/keystatic#1016] [Q&A] [answered]
Hi,
I am using an inline content component in my MDX Field (in an Astro porject) and it does not write the expected markdown. For testing I simply copied the inline component example from the docs and it behaves the same way:

`content: fields.mdx({
          label: "Content",
          components: {
            StatusBadge: inline({
              label: "StatusBadge",
              schema: {
                status: fields.select({
                  label: "Status",
                  options: [
                    { label: "To do", value: "todo" },
                    { label: "In Progress", value: "in-progress" },
                    { label: "Ready for review", value: "ready-for-review" },
                    { label: "Done", value: "done" },
                  ],
                  defaultValue: "todo",
                }),
              },
            }),
          }})`

I expect this output: `<StatusBadge status="in-progress" />` 
What I get is: `<StatusBadge value={{"status":"in-progress"}} extraFiles={[]} />`

Of course we can work around it but... Am I missing something here?

**Comment by emmatown** [comment, marked as answer]
Fixed by #1017
